RestaurantManagementSystem/RestaurantManagementSystem/src/Core/Contracts/IRegistrationRequestRepo.cs

```diff
@@ -0,0 +1,12 @@
+﻿using RestaurantManagementSystem.src.Core.Entities;
+
+namespace RestaurantManagementSystem.src.Core.Contracts
+{
+    public interface IRegistrationRequestRepo
+    {
+        public void AddRequest(RegistrationRequest waitingListRestaurant);
+        public Task<RegistrationRequest> FindByIdAsync (string ID);
+        public void Update(RegistrationRequest registrationRequest);
+        public string GenerateVerificationCode();
```

Comment:
the repository is for the database logic, querying..etc `GenerateCode` is not related

Reply:
then where ? (crying)

Reply:
Create a separate class e.g. `CodeGeneratorService` #singleResponsibility